docs/build/dapp-staking/dapp_staking_architecture.md

```diff
@@ -0,0 +1,672 @@
+
+If users have unclaimed rewards from the previous period or eras, they **NEED** to `claim` them first before staking in the new period.
+- To put into another words, calling `stake` or `unstake` won’t be possible unless **ALL** pending rewards have been claimed;
+- This will help prevent unbounded storage growth and make the overall runtime logic much simpler;
+
+Since there is no `unbonding period` related to unstaking, the `transfer_nomination` call **won’t be** included in the public API. However, the same functionality can be achieved just by batching `[stake, unstake]` calls.
+
+### Claiming Stake Rewards
+
+Stake rewards are rewarded for those who `stake` their tokens on a dApp. Even if tokens are `locked`, they won’t earn any rewards unless they are `staked`. Even though stakes are specific to a dApp, reward claiming will be done for all staked dApps via a single call (this is a change compared to v2 and it will make claiming more efficient & cheaper).
+
+Single call will be able to claim rewards for multiple eras (while retaining **O(1)** complexity).
+
+Unlike v2, there won’t be an option for compounding rewards, at least not directly via an extrinsic call to make compounding rewards work, frontend will need to correctly estimate the reward from claims, and call `lock & stake` extrinsic calls in a batch. The main reason why there’s no intrinsic compounding reward support is that reward claiming isn’t done for a specific contract, but for all of them at once. What dApp would user re-stake their funds on?
+
+The reward for user is calculated using a simple formula:
+
+$user\_reward = total\_stake\_reward * \frac{user\_stake}{total\_staked}$
+
+Rewards are ONLY issued when they are being claimed.
+
+### Claiming Bonus Rewards
+
+If staker staked X **amount** during `voting subperiod` and hasn’t reduced their stake amount below X during `build&earn subperiod`, they will be eligible for extra rewards
+
+$bonus\_reward = total\_bonus\_reward * \frac{user\_contract\_vp\_stake}{total\_vp\_stake}$
+
+The formula demonstrates that bonus reward is calculated per contract and if user was a *loyal staker* for two distinct contracts, they will need to execute two `claim` calls, one for each contract. **vp** refers to `voting subperiod`.
+
+It will only be possible to claim bonus reward for **1 past period**, and only if this period isn’t beyond history depth limit. This shouldn't be a problem, since it's safe to assume that the periods will last longer than a month.
+
+Rewards are ONLY issued when they are being claimed..
+
+### Claiming Dapp Rewards
+
+After an era ends, dApps will be assigned into tiers.
+
+The call footprint is kept the same as in v2, where both smart contract & era need to be specified when calling `claim`.  
+Rewards are ONLY issued when they are being claimed.
```

Comment:
I suggest to remove everything up to this point since it's all covered by the existing docs.

There's no new information.